test/framework/deployment_helpers.go

```diff
@@ -32,7 +32,7 @@ import (
 	. "github.com/onsi/gomega"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/api/policy/v1beta1"
+	policyv1 "k8s.io/api/policy/v1"
```

Comment:
Did some research. PDB was promoted to v1 in Kubernetes v1.21. This means this test now only works on Kubernetes >= v1.21.

As far as I know this test is usually only run against new Kubernetes versions (in core CAPI only v1.25 after this PR)

@fabriziopandini WDYT, should be fine to just use v1 instead of v1beta1?

Reply:
Going by https://cluster-api.sigs.k8s.io/reference/versions.html , we should still allow these tests to run on all supported versions, the minimum management cluster version seems `v1.20` right now?

Reply:
There is no column for v1.3

Reply:
Hm. As we want to cherry-pick this PR into v1.2....
@oscr Can you add an if/else that we're using v1 only for Kubernetes >= v1.21?

Reply:
@sbueringer No problem! Fixed.